packages/valory/skills/decision_maker_abci/behaviours/sampling.py

```diff
@@ -74,32 +64,83 @@ def processable_bet(self, bet: Bet, now: int) -> bool:
 
         within_ranges = within_opening_range and within_safe_range
 
-        # rebetting is allowed only if we have already placed at least one bet in this market.
-        # conversely, if we should not rebet, no bets should have been placed in this market.
-        if self.should_rebet ^ bool(bet.n_bets):
-            return False
+        # check if bet queue number is processable
+        processable_statuses = {
+            QueueStatus.TO_PROCESS,
+            QueueStatus.PROCESSED,
+            QueueStatus.REPROCESSED,
+        }
```

Comment:
Would this cause an exception here?

https://github.com/valory-xyz/trader/blob/3aa2568368673fbae87da8cf0f2433b9c09676e7/packages/valory/skills/decision_maker_abci/behaviours/base.py#L369-L372

Reply:
Theoretically and logically both we will never reach this else check, actually we will never reach this check, but have added it for code readability and understanding

Reply:
What confuses me is that we accept markets with reprocessed status but then in `update_bet_transaction_information` we do not.

Reply:
Yup I have fixed this, we should accept all actually until kpi has been met, you check the next_status function in bets.py